From 26862024487ae101944116d65c352b23592339fb Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Fri, 28 Feb 2020 18:00:40 +0800
Subject: [PATCH] usb: gadget: configfs: fix defined but not used build
 warnings

Fix the following build warnings in configfs driver instead of
scripts/gcc-wrapper.py.

"configfs.c:1488", # drivers/usb/gadget/configfs.c:1488:12: warning: 'configfs_composite_setup' defined but not used
"configfs.c:1513", # drivers/usb/gadget/configfs.c:1513:13: warning: 'configfs_composite_disconnect' defined but not used

Fixes: a77196fc010d ("ANDROID: usb: gadget: configfs: Add Uevent to notify userspace")
Change-Id: I58ef495d6e84af8000d0bd7a3ce3a2ed7947b71f
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/gadget/configfs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/usb/gadget/configfs.c b/drivers/usb/gadget/configfs.c
index 1f59c6c1b38a..247f8ada6c6a 100644
--- a/drivers/usb/gadget/configfs.c
+++ b/drivers/usb/gadget/configfs.c
@@ -1485,6 +1485,7 @@ static void configfs_composite_unbind(struct usb_gadget *gadget)
 	spin_unlock_irqrestore(&gi->spinlock, flags);
 }
 
+#ifndef CONFIG_USB_CONFIGFS_UEVENT
 static int configfs_composite_setup(struct usb_gadget *gadget,
 		const struct usb_ctrlrequest *ctrl)
 {
@@ -1531,6 +1532,7 @@ static void configfs_composite_disconnect(struct usb_gadget *gadget)
 	composite_disconnect(gadget);
 	spin_unlock_irqrestore(&gi->spinlock, flags);
 }
+#endif
 
 static void configfs_composite_suspend(struct usb_gadget *gadget)
 {
-- 
2.35.3

